File: Online-Market-Place-Server/Src/Core/OnlineMarketPlace.Domain/Entities/AttributeType.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace OnlineMarketPlace.Domain
{
    public class AttributeType : Entity
    {
        public string Name { get; set; }
        public DataType DataType { get; set; }

        public ObjectType ObjectType { get; set; }
        public int ObjectTypeId { get; set; }

        public ICollection<AttributeValue> AttributeValues { get; set; }
    }
}

File: Online-Market-Place-Client/src/app/models/attribute-type.ts
import { DataType } from "../enums/data-type";
import { AttributeValue } from "./attribute-value";
import { ProductType } from "./product-type";

export class AttributeType {
    id: number;
    name: string;
    dataType: DataType;
    productType: ProductType;
    attributeValues: AttributeValue[];
}